### PR TITLE
Split An Object into multiple objects

### LINKED ISSUE #5: Split Object
## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [ ] The function's name is clear and helpful
  - [ ] The parameter names are clear and helpful
  - [ ] Default parameters are used if possible
- **JSDoc Comment**
  - [ ] The docstring is well-formatted
  - [ ] There are no spelling mistakes
  - [ ] The description of the function's behavior is clear
  - [ ] All parameters are documented with types
  - [ ] Documented parameter names match the function's parameter names
  - [ ] Parameters have a good description
  - [ ] If default parameters are used, they are the correct type
  - [ ] The return value is documented with a type
  - [ ] The return value has a clear description
- **README**
  - [ ] The documentation is rendered into the README
  - [ ] Docs were generated just before pushing
  - [ ] Docs in the README matches the JSDoc comment
  - [ ] The README contains at least one use case
- **Unit Tests**
  - [ ] Unit tests are present
  - [ ] There are at least 8 test cases
  - [ ] All `describe`s and `it`s have clear and helpful names
  - [ ] All test names actually describe what is being tested
  - [ ] Tests are grouped logically using `describe`
  - [ ] There is only one `expect` inside every `it`
  - [ ] There is no logic (loops, conditionals, ...) in the test cases
  - [ ] Side-effects are tested, if necessary
  - [ ] Tests were run just before pushing
  - [ ] All tests pass (`file-name.pass.txt`)
  - [ ] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [ ] Variables have names that help understand the strategy
  - [ ] Spacing and formatting help to understand the strategy
  - [ ] The code is written as simply and clearly as possible
  - [ ] Comments _are_ used to label important goals in your strategy
  - [ ] Comments _are not_ used to describe how the code works
- **README**
  - [ ] There is an explanation of the solution's strategy in the README
  - [ ] The explanation is clear and understandable
  - [ ] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [ ] The code was formatted just before pushing
  - [ ] No errors occurred while formatting the code
- **Linting**
  - [ ] There should be no linting errors or warnings in the `file-name.lint.txt` file
  - [ ] Spelling should be correct in all code, comments and docstrings
  - [ ] The code was linted just before pushing
- **The solution file**
  - [ ] Has a JSDoc comment
  - [ ] Has an exported function
  - [ ] Does no other code or comments (you can use the sandbox for saving extra ideas)
- **The function body _Should_ ...**
  - [ ] Be well and consistently formatted
  - [ ] Have clear and helpful variable names
  - [ ] Declare variables in the smallest scope possible
  - [ ] Be a simple and readable implementation
  - [ ] Use short comments to describe the strategy
  - [ ] Use short comments to describe important language feature choices
- **The function body should _Not_ ...**
  - [ ] Have extra comments that are not about strategy or important decisions
  - [ ] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [ ] Have any `console` calls (`.log`, `.assert`, ...)
  - [ ] Have any `debugger` statements

### PR DESCRIPTION
**Pull request by mayuri2510**
---
name: Split Object code Review
about: A template PR for code review, complete with checklist
title: 'Code Review: Splits an object into multiple objects with one key/value pair each.'
---

Before the reviewer merges this PR to main/master, all of these boxes must be checked. Only the reviewer can check boxes.

## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [x] The function's name is clear and helpful
  - [x] The parameter names are clear and helpful
  - [x] Default parameters are used if possible
- **JSDoc Comment**
  - [x] The docstring is well-formatted
  - [x] There are no spelling mistakes
  - [x] The description of the function's behavior is clear
  - [x] All parameters are documented with types
  - [x] Documented parameter names match the function's parameter names
  - [x] Parameters have a good description
  - [x] If default parameters are used, they are the correct type
  - [x] The return value is documented with a type
  - [x] The return value has a clear description
- **README**
  - [x] The documentation is rendered into the README
  - [x] Docs were generated just before pushing
  - [x] Docs in the README matches the JSDoc comment
  - [x] The README contains at least one use case
- **Unit Tests**
  - [x] Unit tests are present
  - [x] There are at least 8 test cases
  - [x] All `describe`s and `it`s have clear and helpful names
  - [x] All test names actually describe what is being tested
  - [x] Tests are grouped logically using `describe`
  - [x] There is only one `expect` inside every `it`
  - [x] There is no logic (loops, conditionals, ...) in the test cases
  - [x] Side-effects are tested, if necessary
  - [x] Tests were run just before pushing
  - [x] All tests pass (`file-name.pass.txt`)
  - [x] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [x] Variables have names that help understand the strategy
  - [x] Spacing and formatting help to understand the strategy
  - [x] The code is written as simply and clearly as possible
  - [x] Comments _are_ used to label important goals in your strategy
  - [x] Comments _are not_ used to describe how the code works
- **README**
  - [x] There is an explanation of the solution's strategy in the README
  - [x] The explanation is clear and understandable
  - [x] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [x] The code was formatted just before pushing
  - [x] No errors occurred while formatting the code
- **Linting**
  - [x] There should be no linting errors or warnings in the `file-name.lint.txt` file
  - [x] Spelling should be correct in all code, comments and docstrings
  - [x] The code was linted just before pushing
- **The solution file**
  - [x] Has a JSDoc comment
  - [x] Has an exported function
  - [x] Does no other code or comments (you can use the sandbox for saving extra ideas)
- **The function body _Should_ ...**
  - [x] Be well and consistently formatted
  - [x] Have clear and helpful variable names
  - [x] Declare variables in the smallest scope possible
  - [x] Be a simple and readable implementation
  - [x] Use short comments to describe the strategy
  - [x] Use short comments to describe important language feature choices
- **The function body should _Not_ ...**
  - [x] Have extra comments that are not about strategy or important decisions
  - [x] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [x] Have any `console` calls (`.log`, `.assert`, ...)
  - [x] Have any `debugger` statements

### CODE REVIEW

**Comment by erkamguresen** [comment]
I am reviewing one already. It is better if everybody reviews at least once.

**Comment by OksanaShulha** [comment]
@erkamguresen you are right, @mayuri2510 I'll review yours within an hour

**Comment by OksanaShulha** [comment]
@mayuri2510 I checked every check from you request :) It looks very good but I don't see an explanation of the solution's strategy in the README can you add please

**Comment by mayuri2510** [comment]
> @mayuri2510 I checked every check from you request :) It looks very good but I don't see an explanation of the solution's strategy in the README can you add please

Yes.. sure I add 👍

File: src/split-object/split-object.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    arr.push(obj1);
+    arr.push(obj2);
+    arr.push(obj3);
+
+    return arr;
+  }
+  // console.log("argument should be object and length should be three ");
+  return arr;
+};
```
Comment (OksanaShulha) [inline src/split-object/split-object.js:60]: @mayuri2510 Inside the function body please use short comments to describe the strategy and important language feature choices.

Comment (emelysalmeron) [reply src/split-object/split-object.js:60]: Hello @mayuri2510  after running lint inside of the document **src/split-object/split-object.js** i've found:
2 errors and 2 warnings found.
2 errors and 2 warnings potentially fixable with the `--fix` option.

**Review by emelysalmeron — CHANGES_REQUESTED** [review]

File: src/split-object/split-object.lint-error.txt [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+warning: Delete `␍` (prettier/prettier) at src/split-object/split-object.js:57:3:
+  55 |     return arr;
+  56 |   }
+> 57 | };
+     |   ^
+  58 | 
+
+
+3 errors and 60 warnings found.
+3 errors and 59 warnings potentially fixable with the `--fix` option.
```
Comment (emelysalmeron) [inline src/split-object/split-object.lint-error.txt]: There are a few error lint that need to be fix @mayuri2510

File: src/split-object/README.md [hunk 3]
```diff
@@ -26,3 +26,38 @@ Here's a starter docstring for your solutions. Feel free to rewrite it if that h
...
+
+**Example**
+
+```js
+splitObject({});
+// -> []
+```
+
+> Docs generated: Thu May 20 2021, 11:07:35 AM
+
```
Comment (emelysalmeron) [inline src/split-object/README.md:137]: Hello @mayuri2510 we just missed a few smalls details to develop in the README file such as:
* Strategy
* Implementation
* Use Cases
* Inspiration (if you got inspired in any prev exercise or website that you would like to share)

**Review by emelysalmeron — CHANGES_REQUESTED** [review]

**Comment by mayuri2510** [comment]
@emelysalmeron  @OksanaShulha  As per your requested changes ,I did all and updated files, can you take a look again.

**Review by emelysalmeron — APPROVED** [review]
After reviewing everything seems fine now

**Review by OksanaShulha — APPROVED** [review]
